cabal-install/Distribution/Client/CmdShowBuildInfo.hs

```diff
@@ -0,0 +1,240 @@
+-- | cabal-install CLI command: new-show-build-info
+--
+module Distribution.Client.CmdShowBuildInfo (
+    -- * The @show-build-info@ CLI and action
+    showBuildInfoCommand,
+    showBuildInfoAction
+  ) where
+
+import Distribution.Client.ProjectOrchestration
+import Distribution.Client.CmdErrorMessages
+import Distribution.Client.CmdInstall.ClientInstallFlags
+
+import Distribution.Client.Setup
+         ( GlobalFlags, ConfigFlags(..), ConfigExFlags, InstallFlags )
+import qualified Distribution.Client.Setup as Client
+import Distribution.Simple.Setup
+         ( HaddockFlags, fromFlagOrDefault, TestFlags )
+import Distribution.Simple.Command
+         ( CommandUI(..), usageAlternatives )
+import Distribution.Verbosity
+         ( Verbosity, silent )
+import Distribution.Simple.Utils
+         ( wrapText, die')
+import Distribution.Types.UnitId (UnitId)
+
+import qualified Data.Map as Map
+import qualified Distribution.Simple.Setup as Cabal
+import Distribution.Client.SetupWrapper
+import Distribution.Simple.Program ( defaultProgramDb )
+import qualified Distribution.Client.InstallPlan as InstallPlan
+import Distribution.Client.ProjectPlanning.Types
+import Distribution.Client.ProjectPlanning (
+  setupHsConfigureFlags, setupHsConfigureArgs,
+  setupHsBuildFlags, setupHsBuildArgs, 
+  setupHsScriptOptions
+  )
+import Distribution.Client.DistDirLayout (distBuildDirectory)
+import Distribution.Client.Types ( PackageLocation(..), GenericReadyPackage(..) )
+import Distribution.Client.JobControl (newLock, Lock)
+import Distribution.Simple.Configure (tryGetPersistBuildConfig)
+import Data.List (find)
+
+showBuildInfoCommand :: CommandUI (ConfigFlags, ConfigExFlags, InstallFlags, HaddockFlags, TestFlags)
```

Comment:
What needs to happen either here or in a subsequent PR is to implement the ability to refer to targets by their unique "unit id". So far this command appears to only support the less accurate symbolic target specifiers which aren't able to address all elements of a build-plan.

Reply:
Unit Id as in `TargetInfo.ComponentLocalBuildInfo.componentUnitId`? Or whose unit id?
Are we missing components?